perfkitbenchmarker/providers/gcp/gcp_dpb_dataflow_template.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2022 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module containing class for GCP's Dataflow service.
+Use this module for running Dataflow jobs from pre-built Dataflow templates
+such as https://cloud.google.com/dataflow/docs/guides/templates/provided-templates
+
+No Clusters can be created or destroyed, since it is a managed solution
+See details at: https://cloud.google.com/dataflow/
+"""
+
+import os
+import re
+import time
+import json
+import logging
+import datetime
+
+from absl import flags
+from perfkitbenchmarker import beam_benchmark_helper
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import providers
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.providers.gcp import gcp_dpb_dataflow
+from perfkitbenchmarker.providers.gcp import util
+
+# Refer to flags with prefix 'dpb_df_template' defined in gcp provider flags.py
+FLAGS = flags.FLAGS
+
+class GcpDpbDataflowTemplate(gcp_dpb_dataflow.GcpDpbDataflow):
+  """Object representing GCP Dataflow service for running job templates."""
+
+  CLOUD = providers.GCP
+  SERVICE_TYPE = 'dataflow_template'
+
+  def __init__(self, dpb_service_spec):
+    super(GcpDpbDataflowTemplate, self).__init__(dpb_service_spec)
+    self.project = util.GetDefaultProject()
+    self.input_sub_empty = False
+    self.job_drained = False
+
+  @staticmethod
+  def CheckPrerequisites(benchmark_config):
+    del benchmark_config  # Unused
+    if not FLAGS.dpb_df_template_gcs_location:
+      raise errors.Config.InvalidValue('Template GCS location missing.')
+
+  def Create(self):
+    """See base class."""
+    pass
+
+  def Delete(self):
+    """See base class."""
+    pass
+
+  def SubmitJob(
```

Comment:
How would you add additional arguments like `template_gcs_location`? Also, some of the arguments in the base class are not applicable like `pyspark_file`, should they still be defined here?
Note that in the original parent provider gcp_dpb_dataflow.py, the arguments of JobSubmit() also diverge from the base class.

Reply:
> Note that in the original parent provider gcp_dpb_dataflow.py, the arguments of JobSubmit() also diverge from the base class.

This is an antipattern. The arguments shouldn't change in any child class.  

But, I do think the Dataflow classes stick out like a sore thumb in the DpbService hierarchy, because they are the only ones that run non Hadoop/Spark workloads.

I will let this pass, since there's not any quick solution for this. Just add a TODO for me, reminding me to address the change in signature.

Reply:
Sounds good. TODO added. After this merge, I can also help make the signature consistent across Dataflow classes and BaseDpbService class.